cms/djangoapps/contentstore/views/preview.py

```diff
@@ -124,6 +125,28 @@ def user_id(self):
         return self._request.user.id
 
 
+class StudioPermissionsService(object):
```

Comment:
Also, as I mentioned in the comments, if this commit to check permissions when referencing a library (23e734ba716844f41109831a38d1e5a9abdda7f3) introduces too many open questions, we can schedule it to be merged later as a separate PR. Right now, priority is getting this PR merged ASAP, and I don't think it's really an urgent security concern if users who already have permission to edit courses can use this technique to read the content of libraries they shouldn't. But if we can include it now that would be best, obviously.